### PR TITLE
fix(graph): avoid duplicate state in compiled subgraphs

### LINKED ISSUE #4305: [BUG] data duplication when using APPEND strategy in nested subgraphs
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

When nesting a subgraph within a parent graph using the default APPEND strategy, this issue occurs:

The subgraph inherits the parent's history and returns the full state (history + new items). The parent graph then appends this full return value to its existing history, resulting in exponential data duplication.

### Expected Behavior

The framework should handle subgraph results correctly by avoiding the re-appending of historical data that was already present in the parent state (e.g., by supporting a REPLACE behavior for subgraphs or delta-only updates).

### Steps To Reproduce

run this code:

<details>
<summary><b>source code for reproduce</b></summary>

```java
import com.alibaba.cloud.ai.graph.*;
import com.alibaba.cloud.ai.graph.action.AsyncNodeAction;

import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.CompletableFuture;

public class BugReproducer {

    private static final String KEY_OUTPUT = "output";

    private static final AsyncNodeAction nodeA = state ->
            CompletableFuture.completedFuture(Map.of(KEY_OUTPUT, "a"));

    private static final AsyncNodeAction nodeB = state ->
            CompletableFuture.completedFuture(Map.of(KEY_OUTPUT, "b"));

    private static final AsyncNodeAction nodeC = state ->
            CompletableFuture.completedFuture(Map.of(KEY_OUTPUT, "c"));

    private static final AsyncNodeAction nodeD = state ->
            CompletableFuture.completedFuture(Map.of(KEY_OUTPUT, "d"));

    public static void main(String[] args) throws Exception {

        KeyStrategyFactory appendStrategy = () -> {
            HashMap<String, KeyStrategy> map = new HashMap<>();
            map.put(KEY_OUTPUT, KeyStrategy.APPEND);
            return map;
        };

        KeyStrategyFactory replaceStrategy = () -> {
            HashMap<String, KeyStrategy> map = new HashMap<>();
            map.put(KEY_OUTPUT, KeyStrategy.REPLACE);
            return map;
        };

        // ------- build sub graph A -------
        StateGraph subGraphA = new StateGraph(appendStrategy);

        subGraphA.addNode("a", nodeA);
        subGraphA.addNode("b", nodeB);
        subGraphA.addNode("c", nodeC);

        subGraphA.addEdge(StateGraph.START, "a");
        subGraphA.addEdge("a", "b");
        subGraphA.addEdge("b", "c");
        subGraphA.addEdge("c", StateGraph.END);

        CompiledGraph compiledGraphA = subGraphA.compile();
        // ------- end -------

        // ------ build sub graph B -------
        StateGraph subGraphB = new StateGraph(appendStrategy);

        subGraphB.addNode("c", nodeC);
        subGraphB.addNode("d", nodeD);

        subGraphB.addEdge(StateGraph.START, "c");
        subGraphB.addEdge("c", "d");
        subGraphB.addEdge("d", StateGraph.END);

        CompiledGraph compiledGraphB = subGraphB.compile();
        // ------ end -------

        // ------ build main graph -------
        StateGraph mainGraph = new StateGraph(appendStrategy);

        mainGraph.addNode("a", nodeA);
        mainGraph.addNode("subA", compiledGraphA);
        mainGraph.addNode("subB", compiledGraphB);

        mainGraph.addEdge(StateGraph.START, "a");
        mainGraph.addEdge("a", "subA");
        mainGraph.addEdge("subA", "subB");
        mainGraph.addEdge("subB", StateGraph.END);

        CompiledGraph compiledGraph = mainGraph.compile();
        // ------ end -------

        OverAllState state = compiledGraph.invoke(Collections.emptyMap()).orElseThrow();
        List<Object> output = state.value(KEY_OUTPUT, Collections.emptyList());

        System.out.println("actual output size: " + output.size() + ", which suppose to be 6");
        System.out.println("actual output: " + output);
        System.out.println(", which suppose to be [a, a, b, c, c, d]");

    }

}
```

</details>

### Environment

```markdown
Spring AI Alibaba version(s): 1.1.2.0
JDK version: 17
```

### Debug logs

the code's execution result in my machine:

```
actual output size: 12, which suppose to be 6
actual output: [a, a, a, b, c, a, a, a, b, c, c, d]
, which suppose to be [a, a, b, c, c, d]
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by March-77**
## Summary

- treat a compiled subgraph's final state as an authoritative snapshot instead of applying parent key strategies to it again
- only replace the parent snapshot after a compiled subgraph returns a state map, preserving parent state on subgraph errors
- preserve the parent state's configured key strategies and store when replacing its data
- add regression tests for consecutive compiled subgraphs with an `APPEND` key and for subgraph error handling

## Root cause

A compiled subgraph starts with the parent's full state and returns its full final state. The parent executor then merged that result as a delta, so `APPEND` re-added all inherited values after every subgraph completion.

The snapshot replacement must only run after a successful final map result. Error responses do not contain a state result and must leave the existing parent state unchanged.

## Impact

Nested compiled subgraphs no longer duplicate inherited list entries, and subgraph errors no longer clear the parent state. Ordinary embedded Flux nodes keep their existing incremental merge behavior.

## Validation

- `./mvnw -pl :spring-ai-alibaba-graph-core -Dtest=SubGraphTest,CompiledSubGraphTest,StateGraphTest test` (53 tests, 0 failures/errors, 1 skipped)
- `./mvnw -pl :spring-ai-alibaba-graph-core -Dtest=*,!DatabaseStorePostgreSqlIntegrationTest test` (399 tests, 0 failures/errors, 71 environment-dependent skips)
- Checkstyle: 0 violations

The excluded PostgreSQL integration test requires a running Docker environment.

Fixes #4305

### CODE REVIEW

File: spring-ai-alibaba-graph-core/src/main/java/com/alibaba/cloud/ai/graph/executor/NodeExecutor.java [hunk 1]
```diff
@@ -578,8 +579,13 @@ private Flux<GraphResponse<NodeOutput>> processGraphResponseFlux(
 			combinedUpdateState.putAll(updateState);
 			Optional<InterruptionMetadata> interruptAfterMetadata = interruptAfterForStreaming(context, combinedUpdateState);
 
-			context.mergeIntoCurrentState(partialStateWithoutFlux);
-			context.mergeIntoCurrentState(updateState);
+			if (context.getNodeAction(context.getCurrentNodeId()) instanceof SubCompiledGraphNodeAction) {
+				context.replaceCurrentState(updateState);
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-ai-alibaba-graph-core/src/main/java/com/alibaba/cloud/ai/graph/executor/NodeExecutor.java:599]: Addressed in `2a0b261e3`. Compiled-subgraph snapshot replacement is explicitly gated on both a state-map result and a non-error response, so an error cannot clear the parent state. The regression test now also routes conditionally after the child error and verifies that the preserved parent state selects the recovery branch. `SubGraphTest` passes 14/14, and the graph-core suite passes 405 tests with 71 environment-dependent skips; Checkstyle and Spotless are clean.

**Review by oss-sentinel-ai — COMMENTED** [review]
### Summary
Reviewed the changes in this PR (152 lines changed). The changes appear to be well-structured.

### Observations
- Changes span 152 lines
- Please ensure all edge cases are covered
- Verify test coverage for the modified code paths

---
*Automated review by github-manager-bot*

**Comment by March-77** [comment]
@yuluo-yx 已重新核对当前 head、最新 main 变化、mergeability、CI、reviews 和 review threads；现有评审意见均已验证并清理，未发现需要追加代码修改的问题。烦请在方便时进行维护者审查，谢谢。
